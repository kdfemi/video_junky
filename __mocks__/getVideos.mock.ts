export const videoMock =  {
    "kind":"youtube#searchResult",
    "etag":"2XRyyyjfe4AsHOsGqmKcuAMyUCM",
    "id":{
       "kind":"youtube#video",
       "videoId":"SO1th-Oh39I"
    },
    "snippet":{
       "publishedAt":"2023-10-08T18:00:15Z",
       "channelId":"UCjIuqG1hCxUdLLxOD_uR0vQ",
       "title":"Joh John Florence And Friends At Rocky Point (4K Raw)",
       "description":"Filmed on October 6, 2023 at Rocky Point on the North Shore of Oahu in Hawai'i. Watch as John John Florence goes for a free ...",
       "thumbnails":{
          "default":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/default.jpg",
             "width":120,
             "height":90
          },
          "medium":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/mqdefault.jpg",
             "width":320,
             "height":180
          },
          "high":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/hqdefault.jpg",
             "width":480,
             "height":360
          }
       },
       "channelTitle":"Surfers of Hawaii",
       "liveBroadcastContent":"none",
       "publishTime":"2023-10-08T18:00:15Z"
    }
} as const;

export const getVideosResponseMock = {
    totalSize: 50,
    page: 1,
    size: 10,
    pageCount: 100,
    pages: 5,
    results: [videoMock]
}

export const getVideosResponseEmptyMock = {
    totalSize: 0,
    page: 1,
    size: 10,
    pageCount: 0,
    pages: 0,
    results: []
}