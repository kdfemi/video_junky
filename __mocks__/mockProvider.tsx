/// Mock Provider is a wrapper around @testing-library/react render method
/// This provides us away to mock Provider used app wide

import React, { PropsWithChildren } from 'react';
import { RenderOptions, render as rtlRender } from '@testing-library/react';

type CustomRenderOptions =  RenderOptions;

function render(
    ui: React.ReactElement,
    {
      ...renderOptions
    } = {} as CustomRenderOptions,
  ) {
    function Wrapper({ children }: PropsWithChildren) {
      return <>{children}</>;
    }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}


export * from '@testing-library/react';

export { render };