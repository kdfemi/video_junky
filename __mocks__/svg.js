/// Fallback support for SVG during testing

// eslint-disable-next-line import/no-anonymous-default-export
export default 'svgr-url';
export const ReactComponent = 'div'

