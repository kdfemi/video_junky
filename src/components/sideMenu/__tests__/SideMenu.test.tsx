import '@testing-library/jest-dom';
import SideMenu from ".."
import { render, screen } from "../../../../__mocks__/mockProvider"
jest.mock('next/navigation', () => ({
    useSearchParams: jest.fn(() => ({
        get: jest.fn(() => '10')
    })),
    usePathname: jest.fn(() => 'path'),
    useRouter: jest.fn(() => ({replace: jest.fn()}))
}))

describe('SideMenu', () => {
    it('Should be in document', () => {
        render(<SideMenu/>);
        const sideMenu = screen.getByTestId('sideMenu');
        expect(sideMenu).toBeInTheDocument();
    })
    it('Should match snapshot', () => {
        render(<SideMenu/>);
        const sideMenu = screen.getByTestId('sideMenu');
        expect(sideMenu).toMatchSnapshot();
    })
})