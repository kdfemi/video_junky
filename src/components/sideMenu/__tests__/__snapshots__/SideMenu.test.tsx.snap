// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Should match snapshot 1`] = `
<div
  data-testid="sideMenu"
>
  <button
    class="flex gap-2 md:hidden mt-2 mb-4 bg-junky-yellow text-primary font-medium p-1 rounded-md items-center"
  >
    <svgr-url
      classname="inline"
      height="24"
      width="24"
    />
    <span>
      Show Videos
    </span>
  </button>
  <aside
    class="w-full md:max-w-[530px] md:pr-3 overflow-y-auto space-y-4 md:h-[calc(100vh-64px-2.5rem)] pb-5 md:relative fixed right-0 bottom-0 top-0 px-4 md:left-[unset] md:right-[unset] md:bottom-[unset] md:top-[unset] bg-primary z-40 transition-transform md:transition-none ease-in duration-500 md:block translate-x-[-100vw]"
  >
    <div
      class="sticky top-0 space-y-2 flex flex-col bg-primary pb-3 z-10 md:pt-0 pt-5 items-center gap-3"
    >
      <div
        aria-label="Close Menu"
        class="md:hidden self-end"
        role="button"
        title="Close Menu"
      >
        <svgr-url
          classname="text-white"
        />
      </div>
      <div
        class="w-full"
      >
        <div
          class="relative flex flex-1 flex-shrink-0"
        >
          <label
            class="sr-only"
            for="search"
          >
            Search
          </label>
          <input
            class="focus:border-junky-yellow block w-full rounded-md border bg-primary text-white border-gray-500 py-[9px] pl-10 text-sm outline-none placeholder:text-gray-500 "
            placeholder="Search Videos"
            value="10"
          />
        </div>
      </div>
    </div>
  </aside>
</div>
`;
